test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/BaseE2E.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	/* This class acts as the base class for all the language test case classes.
+	* Takes care of Initial Orchestration and actual flow of the test.
+	*/
+	public class BaseE2E
+	{
+		private readonly KafkaE2EFixture _kafkaE2EFixture;
+		private readonly Language _language;
+		private readonly BrokerType _brokerType;
+		private readonly E2ETestInvoker _invoker;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		protected BaseE2E(KafkaE2EFixture kafkaE2EFixture, Language language, BrokerType brokerType, ITestOutputHelper output)
+		{
+			_kafkaE2EFixture = kafkaE2EFixture;
+
+			_language = language;
+			_kafkaE2EFixture.SetLanguage(language);
+
+			_brokerType = brokerType;
+			_kafkaE2EFixture.SetBrokerType(brokerType);
+
+			_invoker = new E2ETestInvoker();
+
+			_kafkaE2EFixture.OrchestrateInitialization();
+		}
+
+		public async Task Test(AppType appType, InvokeType invokeType, HttpRequestEntity httpRequestEntity,
+			KafkaEntity queueEntity, List<string> expectedOutput)
+		{
+			//Send invocation Http request to the function apps 
+			await InvokeE2ETest(appType, invokeType, httpRequestEntity, queueEntity);
+
+			// wait for the function completion
+			await Task.Delay(60000);
```

Comment:
Can this be polled by some status ?

Reply:
Not by anyway we are currently aware of. We can take up an investigation on this for later. Thanks for the suggestion.